tests/fe/fe_nedelec_singularity_01.cc

```diff
@@ -0,0 +1,798 @@
+      const double     delta = dimension_x / 1000.;
+      const Point<dim> point_a(dimension_x / 3. - delta, delta, delta);
+      const Point<dim> point_b(dimension_x / 3. + delta, delta, delta);
+      deallog << " Point_a = " << point_a << std::endl;
+      deallog << " Point_b = " << point_b << std::endl;
+      Vector<std::complex<double>> solution_a(dim);
+      Vector<std::complex<double>> solution_b(dim);
+      solution_a = 0;
+      solution_b = 0;
+      {
+        bool point_in_locally_owned_cell;
+        auto mapping = StaticMappingQ1<dim>::mapping;
+        // find the cell in which this point
+        // is, initialize a quadrature rule with
+        // it, and then a FEValues object
+        const std::pair<typename DoFHandler<dim>::active_cell_iterator,
+                        Point<dim>>
+          cell_point = GridTools::find_active_cell_around_point(mapping,
+                                                                dof_handler,
+                                                                point_a);
+
+        point_in_locally_owned_cell = cell_point.first->is_locally_owned();
+        if (point_in_locally_owned_cell)
+          {
+            VectorTools::point_value(dof_handler,
+                                     locally_relevant_solution,
+                                     point_a,
+                                     solution_a);
+          }
+      }
+      {
+        bool point_in_locally_owned_cell;
+        auto mapping = StaticMappingQ1<dim>::mapping;
+        // find the cell in which this point
+        // is, initialize a quadrature rule with
+        // it, and then a FEValues object
+        const std::pair<typename DoFHandler<dim>::active_cell_iterator,
+                        Point<dim>>
+          cell_point = GridTools::find_active_cell_around_point(mapping,
+                                                                dof_handler,
+                                                                point_b);
+
+        point_in_locally_owned_cell = cell_point.first->is_locally_owned();
+        if (point_in_locally_owned_cell)
+          {
+            VectorTools::point_value(dof_handler,
+                                     locally_relevant_solution,
+                                     point_b,
+                                     solution_b);
+          }
+      }
+      // Only one process has the solution_a or/and solution_b. This is a simple
+      // approach to send solution_a and solution_b to all the processes.
+      Utilities::MPI::sum(solution_a, mpi_communicator, solution_a);
+      Utilities::MPI::sum(solution_b, mpi_communicator, solution_b);
+      deallog << " Solution(point_a) : " << solution_a << std::endl;
+      deallog << " Solution(point_b) : " << solution_b << std::endl;
+      // Vector does not provide operator-
+      deallog << " Solution(point_b) - solution (point_a): "
+              << (solution_b -= solution_a) << std::endl;
```

Comment:
`point_a` and `point_b` are very close, but they are in two different elements. The difference between the solution at those two points should be small.

Reply:
Yes.